test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -240,13 +244,44 @@ public final void ensureCleanedUp() throws Exception {
         checkStaticState();
     }
 
+    // Log4j will write out status messages indicating problems with the Log4j usage to the status logger; we hook into this logger and
+    // assert that no such messages were written out as these would indicate a problem with our logging configuration
+    private static final List<StatusData> statusData = new ArrayList<>();
+    private static final StatusConsoleListener listener = new StatusConsoleListener(Level.WARN) {
```

Comment:
super nit: we don't need this static variable, it's only used in the static block, so it could become a local variable there.